tests/sklearn/test_sklearn_models.py

```diff
@@ -1765,3 +1863,54 @@ def test_linear_models_have_no_tlu(
 
     # Check that no TLUs are found within the MLIR
     check_circuit_has_no_tlu(fhe_circuit)
+
+
+@pytest.mark.parametrize("model_class, parameters", get_sklearn_tree_models_and_datasets())
+@pytest.mark.parametrize("n_bits", [2, 5, 11])
+def test_rounding_consistency_for_regular_models(
+    model_class,
+    parameters,
+    n_bits,
+    load_data,
+    check_r2_score,
+    check_accuracy,
+    is_weekly_option,
+    verbose=True,
+):
+    """Test that Concrete ML without and with rounding are 'equivalent'."""
+
+    if verbose:
+        print("Run check_rounding_consistency")
+
+    model = instantiate_model_generic(model_class, n_bits=n_bits)
+
+    # According to https://arxiv.org/pdf/2010.04804.pdf, there are two levels of comparison for
+    # trees: one at the level of X.A < B, and another at the level of I.C == D.
+    # To trigger the rounding in the second level, we need datasets with larger sizes and higher
+    # quantization precision are required.
+    # However, the test will be executed exclusively with rounding enabled, as tree-based models
+    # without rounding cannot be compiled.
+    if n_bits > 9:
```

Comment:
you've defined `N_BITS_MAXIMUM_THRESHOLD_FOR_TREE_BASED_MODELS`, so you need to use it here as well !

Reply:
This part will be removed, since we agreed to test level 2 separately